Upstream-Status: Backport

This fixes the build with WebRTC support turned off.

Signed-off-by: Raphael Kubo da Costa <raphael.kubo.da.costa@intel.com>
---
From 66e6a51773c74189058e55c8b7e63353b4f469f8 Mon Sep 17 00:00:00 2001
From: hlundin <hlundin@chromium.org>
Date: Thu, 22 Jun 2017 08:06:48 -0700
Subject: [PATCH] WebrtcAudioPrivateSetAudioExperimentsFunction dummy
 implementation

This change introduces a dummy implementation of
WebrtcAudioPrivateSetAudioExperimentsFunction::RunAsync() which is
invoked when Chrome is build without support for WebRTC
(enable_webrtc=false).

BUG=730019,708475

Review-Url: https://codereview.chromium.org/2949563002
Cr-Commit-Position: refs/heads/master@{#481532}
---
 .../extensions/api/webrtc_audio_private/webrtc_audio_private_api.cc | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/chrome/browser/extensions/api/webrtc_audio_private/webrtc_audio_private_api.cc b/chrome/browser/extensions/api/webrtc_audio_private/webrtc_audio_private_api.cc
index 8f3fc4c..bc5a07c 100644
--- a/chrome/browser/extensions/api/webrtc_audio_private/webrtc_audio_private_api.cc
+++ b/chrome/browser/extensions/api/webrtc_audio_private/webrtc_audio_private_api.cc
@@ -302,6 +302,7 @@ WebrtcAudioPrivateSetAudioExperimentsFunction::
     ~WebrtcAudioPrivateSetAudioExperimentsFunction() {}
 
 bool WebrtcAudioPrivateSetAudioExperimentsFunction::RunAsync() {
+#if BUILDFLAG(ENABLE_WEBRTC)
   DCHECK_CURRENTLY_ON(BrowserThread::UI);
   std::unique_ptr<wap::SetAudioExperiments::Params> params(
       wap::SetAudioExperiments::Params::Create(*args_));
@@ -320,6 +321,11 @@ bool WebrtcAudioPrivateSetAudioExperimentsFunction::RunAsync() {
 
   SendResponse(true);
   return true;
+#else
+  SetError("Not supported");
+  SendResponse(false);
+  return false;
+#endif
 }
 
 }  // namespace extensions
-- 
2.9.4
